cpp/src/cider/benchmarks/StringHashMapBenchmark.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright(c) 2022-2023 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// #pragma GCC optimize(1)
+// #pragma GCC optimize(2)
+// #pragma GCC optimize(3, "Ofast", "inline")
+
+#include <random>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+#include <common/base/StringRef.h>
+#include <common/hashtable/HashTableAllocator.h>
+#include <common/hashtable/StringHashMap.h>
+#include <folly/container/F14Map.h>
+#include <tests/utils/Utils.h>
+
+using namespace cider::hashtable;
+
+enum GeneratePattern { Sequence, Random };
+
+size_t s_str_len = 10;
+size_t l_str_len = 500;
+
+#define N_MAX std::numeric_limits<T>::max()
+#define N_MIN std::numeric_limits<T>::min()
+
+#define NO_INLINE __attribute__((__noinline__))
+
+static std::string sequence_string(size_t length, size_t index) {

```

Comment:
this function will generate fixed string rather than random string? if so hash map will only store (10+26+26) unique value?

Reply:
yes